src/combinatorics/simple_graph/regularity/uniform.lean

```diff
@@ -130,10 +217,27 @@ end
 
 variables {P G}
 
+lemma is_uniform.mono {ε ε' : 𝕜} (hP : P.is_uniform G ε) (h : ε ≤ ε') : P.is_uniform G ε' :=
+((nat.cast_le.2 $ card_le_of_subset $ P.non_uniforms_mono G h).trans hP).trans $
+  mul_le_mul_of_nonneg_left h $ nat.cast_nonneg _
+
 lemma is_uniform_of_empty (hP : P.parts = ∅) : P.is_uniform G ε :=
 by simp [is_uniform, hP, non_uniforms]
 
 lemma nonempty_of_not_uniform (h : ¬ P.is_uniform G ε) : P.parts.nonempty :=
 nonempty_of_ne_empty $ λ h₁, h $ is_uniform_of_empty h₁
 
+variables (P G ε) (s : finset α)
+
+/-- The witnesses of non-uniformity among the parts of a finpartition. -/
```

Comment:
This isn't giving all witnesses, right? just a system of arbitrary witnesses for the particular partition?
```suggestion
/-- A choice for witnesses of non-uniformity among the parts of a finpartition
using `simple_graph.nonuniform_witness`. -/
```